common/djangoapps/microsite_configuration/__init__.py

```diff
@@ -1 +1,19 @@
+from django.conf import settings as base_settings
+
+from microsite_configuration import microsite
 from .templatetags.microsite import page_title_breadcrumbs
+
+
+class MicrositeAwareSettings():
+    """
+    This class is a handy utility to make a call to the settings
+    completely microsite aware by replacing the:
+    from django.conf import settings
+    with:
+    from microsite_configuration import settings
+    """
+
+    def __getattr__(self, name):
+        return microsite.get_value(name, base_settings.__getattr__(name))
+
+settings = MicrositeAwareSettings()
```

Comment:
I like this very much. We struggle where developers don't set up their calls to use microsite.get_value(), this approach at leasts allows the python to remain consistent: settings.MY_KEY. We then just need to make sure the imports are updated appropriately.